plugin/plugin.go

```diff
@@ -0,0 +1,249 @@
+}
+
+func (p *FlywayPlugin) Run() error {
+	var stdoutBuf, stderrBuf bytes.Buffer
+	var err error
+
+	p.ExecCommand = p.GetExecArgsStr()
+	fmt.Println("Command: ", p.ExecCommand)
+
+	cmdParts := strings.Fields(p.ExecCommand)
+	if len(cmdParts) < 2 {
+		return fmt.Errorf("Invalid command: %s", p.ExecCommand)
+	}
+	cmdName := cmdParts[0]
+	cmdArgs := cmdParts[1:]
+
+	cmd := exec.Command(cmdName, cmdArgs...)
+	cmd.Stdout = &stdoutBuf
+	cmd.Stderr = &stderrBuf
+	if len(p.InputArgs.DriverPath) > 0 {
+		p.Env = "CLASSPATH=" + p.InputArgs.DriverPath
+		cmd.Env = append(os.Environ(), "CLASSPATH="+p.InputArgs.DriverPath)
+	}
+
+	if p.InputArgs.IsDryRun == "TRUE" {
+		return nil
+	}
+
+	err = cmd.Run()
+	if err == nil {
+		fmt.Println(stdoutBuf.String())
+		fmt.Printf("Command execution success")
+	} else {
+		fmt.Println(stderrBuf.String())
+		fmt.Printf("Error executing command: %v\n", err.Error())
+	}
+
+	return nil
+}
+
+func (p *FlywayPlugin) GetExecArgsStr() string {
+	var execCommand string
+
+	execCommand += GetFlywayExecutablePath() + " "
+	execCommand += p.InputArgs.FlywayCommand + " "
+	execCommand += p.CommandSpecificArgs + " "
+
+	if len(p.InputArgs.Url) > 0 {
+		execCommand += "-url=" + p.InputArgs.Url + " "
+	}
+	if len(p.InputArgs.UserName) > 0 {
+		execCommand += "-user=" + p.InputArgs.UserName + " "
+	}
+	if len(p.InputArgs.Password) > 0 {
+		execCommand += "-password=" + p.InputArgs.Password + " "
+	}
+	if len(p.InputArgs.Locations) > 0 {
+		execCommand += "-locations=" + p.InputArgs.Locations + " "
+	}
+	// this should be the last
+	execCommand += p.InputArgs.CommandLineArgs
+
+	return execCommand
+}
+
+func (p *FlywayPlugin) IsCommandValid() error {
+	if p.InputArgs.FlywayCommand == "" {
+		return fmt.Errorf("Command is empty")
+	}
+
+	err := p.IsUnknownCommand()
+	if err != nil {
+		return err
+	}
+
+	err = p.CheckMandatoryArgs()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *FlywayPlugin) IsUnknownCommand() error {
+	_, ok := knownCommandsMap[p.InputArgs.FlywayCommand]
+	if ok {
+		return nil
+	}
+
+	return fmt.Errorf("Unknown command: %s", p.InputArgs.FlywayCommand)
+}
+
+func (p *FlywayPlugin) CheckMandatoryArgs() error {
+
+	args := p.InputArgs
+
+	if strings.Contains(args.CommandLineArgs, ConfigFileOpt) { // pick args from file
+		return nil
+	}
+
+	type mandatoryArg struct {
+		EnvName   string
+		ParamName *string
+		Hint      string
+	}
+
+	ma := []mandatoryArg{
+		{"FLYWAY_URL", &args.Url, "url"},
+		{"FLYWAY_USER", &args.UserName, "username"},
+		{"FLYWAY_PASSWORD", &args.Password, "password"},
+	}
+
+	for _, m := range ma {
+		if os.Getenv(m.EnvName) == "" && *m.ParamName == "" {
+			LogPrintln("Missing mandatory argument: " + m.EnvName)
+			return fmt.Errorf("Missing mandatory argument: %s", m.Hint)
+		}
+	}
+
+	return nil
+}
+
+var knownCommandsMap = map[string]bool{
+	MigrateCommand:  true,
+	CleanCommand:    true,
+	BaselineCommand: true,
+	RepairCommand:   true,
+	ValidateCommand: true,
+}
```

Comment:
instead of map[string]bool use a map[string]string to associate commands with descriptions for clarity - eg: `MigrateCommand:  "Performs database migration",`

Reply:
done